### PR TITLE
Fixed intersections of primitive unions with `exactOptionalPropertyTypes` when mixing `undefinedType` with `missingType`

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/58174

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot test it

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -17430,6 +17436,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
...
                 if (insertType(checked, t)) {
                     if (eachUnionContains(unionTypes, t)) {
+                        // undefinedType/missingType should always come sorted first so we leverage that here
+                        if (t === undefinedType && result.length && result[0] === missingType) {
+                            continue;
+                        }
+                        if (t === missingType && result.length && result[0] === undefinedType) {
+                            result[0] = missingType;
+                            continue;
+                        }
```
Comment (jakebailey) [inline src/compiler/checker.ts:17446]: I used this PR as test for some new code coverage stuff, and it actually revealed that this code is never tested: https://app.codecov.io/gh/jakebailey/TypeScript-codecov-testing/pull/3

Comment (Andarist) [reply src/compiler/checker.ts]: I pushed out a new test case and verified manually that those lines are hit now

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -17839,6 +17839,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function eachUnionContains(unionTypes: UnionType[], type: Type) {
         for (const u of unionTypes) {
             if (!containsType(u.types, type)) {
+                if (type === missingType) {
```
Comment (andrewbranch) [inline src/compiler/checker.ts:17842]: Can you update the comment above this function?

**Review by andrewbranch — APPROVED** [review]

**Comment by andrewbranch** [comment]
@typescript-bot pack this

**Pull request merged**